coalib/testing/LocalBearTestHelper.py

```diff
@@ -52,6 +54,22 @@ class LocalBearTestHelper(unittest.TestCase):
     If you miss some methods, get in contact with us, we'll be happy to help!
     """
 
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        self.assertEqual(len(observed_result), len(expected_result),
+                         msg='The number of elements in the observed result '
+                             'and expected result differ.')
+        for observed, expected in zip_longest(
+                observed_result, expected_result):
+            if (isinstance(observed, Comparable)
+                and isinstance(expected, Comparable)) and (
+                        type(observed) is type(expected)):
+                for attribute in type(observed).__compare_fields__:
+                    self.assertEqual(getattr(observed, attribute),
+                                     getattr(expected, attribute),
+                                     msg='{} mismatch.'.format(attribute))
```

Comment:
again, details of the differences are lost.

There is also a problem with iterating and using assertEqual, that the first attribute of the first message to be different will cause the test to fail and no further details being shown.  That isnt helpful if there are more than one difference in the output vs expected, which is extremely common.
Also the order of the attributes compared needs to be considered if the first difference is going to be the only mismatch shown.  The ordering for `Result` is currently:

1. 'affected_code'
2. 'severity'
3. 'confidence'
4. 'origin'
5. 'message_base'
6. 'message_arguments'
7. 'aspect'
8. 'additional_info'
9. 'debug_msg'

The better implementation of something like this is to do all of the comparisons first, and then if there are any differences an attempt is made to show them sanely to the user.